### PR TITLE
CUDA 12.6.2 windows build

### PR DESCRIPTION
**Pull request by tinglvv**
add windows cuda 12.6 build script, reference https://github.com/pytorch/builder/pull/1725/
keeping same windows driver due to CUDA Minor Version Compatibility

related to https://github.com/pytorch/pytorch/issues/138440

cc @atalman @malfet @ptrblck @nWEIdia

### CODE REVIEW

**Comment by tinglvv** [comment]
Need:
Upload https://developer.download.nvidia.com/compute/cuda/12.6.2/local_installers/cuda_12.6.2_560.94_windows.exe

File: conda/pytorch-nightly/bld.bat [hunk 1]
```diff
@@ -31,6 +31,9 @@ if "%desired_cuda%" == "12.1" (
 if "%desired_cuda%" == "12.4" (
     set TORCH_NVCC_FLAGS=-Xfatbin -compress-all --threads 2
 )
+if "%desired_cuda%" == "12.6" (
+    set TORCH_NVCC_FLAGS=-Xfatbin -compress-all --threads 2
+)
```
Comment (malfet) [inline conda/pytorch-nightly/bld.bat:36]: Is it possible to avoid code duplication here and modify batch script to match all CUDA-12.X?

File: windows/internal/build_magma.bat [hunk 2]
```diff
@@ -35,6 +35,9 @@ cd magma
 mkdir build && cd build
 
 set GPU_TARGET=All
+if "%CUVER_NODOT%" == "126" (
+  set CUDA_ARCH_LIST=-gencode arch=compute_50,code=sm_50 -gencode arch=compute_60,code=sm_60 -gencode arch=compute_70,code=sm_70 -gencode arch=compute_80,code=sm_80 -gencode arch=compute_86,code=sm_86 -gencode arch=compute_90,code=sm_90
+)
```
Comment (malfet) [inline windows/internal/build_magma.bat:40]: Can we modify 12.4 condition to match both 12.4 and 12.6

File: .github/workflows/build-magma-windows.yml [hunk 3]
```diff
@@ -32,7 +32,7 @@ jobs:
       - name: Build MAGMA and push to S3
         run: windows/internal/build_magma.bat
       - name: Save as artifact
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v4
```
Comment (malfet) [inline .github/workflows/build-magma-windows.yml:35]: This change is good, but do you mind move it to the separate PR?

File: windows/internal/driver_update.bat [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
+set "DRIVER_DOWNLOAD_LINK=https://ossci-windows.s3.amazonaws.com/%WIN_DRIVER_VN%-data-center-tesla-desktop-winserver-2016-2019-2022-dch-international.exe"
+curl --retry 3 -kL %DRIVER_DOWNLOAD_LINK% --output %WIN_DRIVER_VN%-data-center-tesla-desktop-winserver-2016-2019-2022-dch-international.exe
 if errorlevel 1 exit /b 1
 
-start /wait 528.89-data-center-tesla-desktop-winserver-2016-2019-2022-dch-international.exe -s -noreboot
+start /wait %WIN_DRIVER_VN%-data-center-tesla-desktop-winserver-2016-2019-2022-dch-international.exe -s -noreboot
 if errorlevel 1 exit /b 1
 
-del 528.89-data-center-tesla-desktop-winserver-2016-2019-2022-dch-international.exe || ver > NUL
+del %WIN_DRIVER_VN%-data-center-tesla-desktop-winserver-2016-2019-2022-dch-international.exe || ver > NUL
```
Comment (malfet) [inline windows/internal/driver_update.bat:9]: Unless there is strong need let's skip driver update here

File: .github/workflows/build-magma-windows.yml (hunk 3 above)
Comment (tinglvv) [reply .github/workflows/build-magma-windows.yml]: Thanks! It appears that this maybe needed, as test is failing with "This request has been automatically failed because it uses a deprecated version of `actions/upload-artifact: v2`" if unchanged

File: .github/workflows/build-magma-windows.yml [hunk 5]
```diff
@@ -32,7 +32,7 @@ jobs:
...
         with:
           path: magma_*_cuda*_*.7z
```
Comment (atalman) [inline .github/workflows/build-magma-windows.yml:37]:
```suggestion
          path: magma_*_cuda*_*.7z
          overwrite: true
```

Comment (atalman) [reply .github/workflows/build-magma-windows.yml:37]: @tinglvv please try this parameter

**Review by atalman — APPROVED** [review]
lgtm
